### DISCUSSION TITLE
Handling Overpressure

### DISCUSSION THREAD

**Discussion by softwareengineerprogrammer** [NatLabRockies/GEOPHIRES-X#176] [Ideas]
Starting this discussion as an offshoot from @malcolm-dsider's comment on https://github.com/NREL/GEOPHIRES-X/issues/167?title=Pump+power+higher+but+net+power+remains+the+same?#issuecomment-2016550370:

> While we are looking at pressure and pumping, I should also point out that perhaps we should be looking at improving how the pressures are handled in order to account for overpressure systems… The DOE just announced $8B new money (yes, that is the word that starts with a B!) for new renewable energy projects in Texas and Louisiana, so the interest in doing geothermal in the overpressure belts of TX and LA is going to go WAY up, as they are among the most prospective targets in the region. As far as I know, GEOPHIRES doesn’t address overpressure. If that is true, I suggest we add something that does. The net effect of overpressure is to make production pumping easier (or zero) while increasing the pumping effort required for injection (this assumes you are injecting back into the source formation – we also might want to figure out how to account for injecting into a different (shallower, lower-pressure) formation). The other characteristic of overpressure I think we should try to account for is pressure depletion – as you allow the overpressure to lift your brine to the surface, it eventually depletes, and you have to start pumping.

> My approach to this would be to add two new parameters to the code. The first one contains the percent of pressure at which the initial overpressure starts that is above and beyond what would be normally expected if the formation was not overpressured  – “Initial overpressure is 30% than expected lithostatic/hydrostatic pressure.” The second new parameter would capture the rate of depletion – “overpressure depletes at 10% per year.” The algorithm can figure out at what point the overpressure drops to the point that the flow rate from the formation is less than the desired flow rate, and it can start adding in production pumping from that point in an ever-increasing amount until it gets to the pumping power you would expect if there never was overpressure (i.e., full depletion). The injection side story is the opposite – we need to calculate how much additional pumping is needed to overcome the overpressure, but then that number goes down as the overpressure depletes.

> In the end, I don’t think this is going to make that big a difference – the uplift you get from the overpressure is going to be close to the increased pumping required to do the injection, but pumping down is always easier than pumping up, so there will be some difference… and in any case, I think our users want to be able to point to evidence in the output from GEOPHIRES that shows it is at least accounting for overpressure, even if the impact turns out to be relatively small.

> If everyone agrees that this work is worthy and that I have the right approach, I’ll put it on my to-do list.

**Comment by softwareengineerprogrammer** [comment]
I don't have any technical insight on this from a geothermal standpoint, but it certainly makes sense conceptually to add support for aspects where substantial DOE funding is a relevant factor. (I'll leave it to @kfbeckers or other expert to comment on geothermal technical aspects.)

Re: the addition of new parameters, my one input is a comment on general principle that we should confirm that those parameters are:
1. Not in conflict with any other existing parameters or calculations
2. Not redundant with existing parameters
3. Not derivable from other parameters/state calculations 

(Again, I have no specific technical insight on overpressure specifically and am just listing these items as a general rubric for evaluating your proposed approach)

**Comment by malcolm-dsider** [comment]
Work completed by PR 185

**Reply by softwareengineerprogrammer** [reply]
Link for posterity: https://github.com/NREL/GEOPHIRES-X/pull/185
